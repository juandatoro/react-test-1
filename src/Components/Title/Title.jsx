import { createElement } from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';

export const Title = ({ type = 'h1', text = '', className, children }) =>
  createElement(type, { className: cn('title', className) }, children ?? text);

Title.propTypes = {
  type: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),
  text: PropTypes.string,
  children: PropTypes.string,
};
