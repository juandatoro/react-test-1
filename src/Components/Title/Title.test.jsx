import { render } from '@testing-library/react';
import { Title } from './Title';

describe('Title component', () => {
  const { container } = render(<Title />);

  test('To render main title and submit btn', () => {
    expect(container).toMatchSnapshot();
  });
});
