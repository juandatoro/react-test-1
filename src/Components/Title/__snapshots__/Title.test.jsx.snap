// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component To render main title and submit btn 1`] = `
<div>
  <h1
    class="title"
  />
</div>
`;
